.github/workflows/test-lang-rust-ci.yml

```
@@ -58,20 +58,59 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Cache Cargo
+      - name: Cache Cargo for 1.65.0
+        if: matrix.rust == '1.65.0' && matrix.target == 'x86_64-unknown-linux-gnu'
         uses: actions/cache@v3
         with:
           # these represent dependencies downloaded by cargo
           # and thus do not depend on the OS, arch nor rust version.
           path: ~/.cargo
-          key: cargo-cache1-
+          key: cargo-cache1-1.65.0-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache Cargo for 1.65.0 (Restore Only)
+        if: matrix.rust == '1.65.0' && matrix.target != 'x86_64-unknown-linux-gnu'
+        uses: actions/cache/restore@v3
+        with:
+          # these represent dependencies downloaded by cargo
+          # and thus do not depend on the OS, arch nor rust version.
+          path: ~/.cargo
+          key: cargo-cache1-1.65.0-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache Cargo for 1.70.0+
+        if: matrix.rust != '1.65.0' && matrix.target == 'x86_64-unknown-linux-gnu'
+        uses: actions/cache@v3
+        with:
+          # these represent dependencies downloaded by cargo
+          # and thus do not depend on the OS, arch nor rust version.
+          path: ~/.cargo
+          key: cargo-cache1-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache Cargo for 1.70.0+ (Restore Only)
+        if: matrix.rust != '1.65.0' && matrix.target != 'x86_64-unknown-linux-gnu'
+        uses: actions/cache/restore@v3
+        with:
+          # these represent dependencies downloaded by cargo
+          # and thus do not depend on the OS, arch nor rust version.
+          path: ~/.cargo
+          key: cargo-cache1-${{ hashFiles('**/Cargo.lock') }}
+
       - name: Cache Rust dependencies
+        if: matrix.target == 'x86_64-unknown-linux-gnu'
         uses: actions/cache@v3
         with:
           # these represent compiled steps of both dependencies and avro
           # and thus are specific for a particular OS, arch and rust version.
-          path: ~/target
-          key: ${{ runner.os }}-target-cache1-${{ matrix.rust }}-
+          path: lang/rust/target
```

IMO this should be 
```suggestion
          path: ./target
```
but it has to be tested!

---

Actually, I have already tried `./target` first. But it doesn't work. `defaults.run.working-directory` seems not to affect here.